mitodl/ocw-hugo-themes
# upgraded webpack

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #778 

#### What's this PR do?
This PR updates webpack module

#### How should this be manually tested?
- Optional: Install `n` through npm (node version manager)
- Upgrade node to v18
- Verify npm run start:course works as expected
- Downgrade node to v16
- Verify npm run start:course works as expected

#### Any Background Context ? 
Upgraded webpack and peer dependencies to latest version. There are a few breaking changes in the following packages
- Expose Loader
- MiniCssExtractor
In the case of expose-loader the API Schema was changed and This PR webpack config to match the new config. For MiniCSSLoader the following options were removed from plugin and loader constructor.
 - MiniCssExtractor.loader
      - filename
      - chunkFileName
      - hmr
 - MiniCSSExtractor.Plugin
     - publicPath
